samples/ei_ncs/src/main.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <ei_ncs.h>
+
+#include "input_data.h"
+
+static size_t frame_surplus;
+
+void result_ready_cb(int err)
+{
+	if (err) {
+		printk("Result ready callback returned error (err: %d)\n", err);
+		return;
+	}
+
+	const char *label;
+	float value;
+	float anomaly;
+
+	int error = ei_ncs_get_classification_results(&label, &value, &anomaly);
+
+	if (error) {
+		printk("Cannot get classification results (err: %d)", err);
+	} else {
+		printk("\nClassification results\n");
+		printk("======================\n");
+		printk("Label: %s\n", label);
+		printk("Value: %.2f\n", value);
+		if (ei_ncs_classifier_has_anomaly()) {
+			printk("Anomaly: %.2f\n", anomaly);
+		}
+	}
+
+	if (frame_surplus > 0) {
+		err = ei_ncs_start_prediction(0, 1);
+		if (err) {
+			printk("Cannot restart prediction (err: %d)\n", err);
+		} else {
+			printk("Prediction restarted...\n");
+		}
+
+		frame_surplus--;
+	}
+}
+
+void main(void)
+{
+	int err = ei_ncs_init(result_ready_cb);
+
+	if (err) {
+		printk("Edge Impulse wrapper failed to initialize (err: %d)\n",
+		       err);
+		return;
+	};
+
+	if (ARRAY_SIZE(input_data) < ei_ncs_get_window_size()) {
+		printk("Not enough input data\n");
+		return;
+	}
+
+	if (ARRAY_SIZE(input_data) % ei_ncs_get_frame_size() != 0) {
+		printk("Imporper number of input samples\n");
+		return;
+	}
+
+	frame_surplus = (ARRAY_SIZE(input_data) - ei_ncs_get_window_size())
+			/ ei_ncs_get_frame_size();
+
+	/* input_data is defined in input_data.h file. */
+	err = ei_ncs_add_data(input_data, ARRAY_SIZE(input_data));
```

Comment:
could we do the loop here something like that:
```
/* Schedule prediction when all require samples are received. */
ei_ncs_start_prediction(on_next_full_window);
while (cnt) {
   /* Simulate sample acquisition */
   ei_ncs_add_data(input_data[cnt], frame_size);
   cnt += frame_size;
   ksleep(xyz);
}
```

and in callback
```
...
/* Schedule prediction on next full window of data. */
ei_ncs_start_prediction(next_window);
```

This way we could show how wrapper can store data but also be able to trigger when enough data is received.

Reply:
Ok, that could be even better in that case

Reply:
Data for the first prediction window is added at once, the additional data is split into frames.
In that way we show both possibilities.